SignalServiceKit/src/Storage/OWSMediaGalleryFinder.m

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSMediaGalleryFinder.h"
+#import "OWSStorage.h"
+#import "TSAttachmentStream.h"
+#import "TSMessage.h"
+#import "TSThread.h"
+#import <YapDatabase/YapDatabaseAutoView.h>
+#import <YapDatabase/YapDatabaseTransaction.h>
+#import <YapDatabase/YapDatabaseViewTypes.h>
+#import <YapDatabase/YapWhitelistBlacklist.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString *const OWSMediaGalleryFinderExtensionName = @"OWSMediaGalleryFinderExtensionName";
+
+@implementation OWSMediaGalleryFinder
+
+#pragma mark - Public Finder Methods
+
+-  (NSUInteger)mediaCountForThread:(TSThread *)thread transaction:(YapDatabaseReadTransaction *)transaction
+{
+    NSString *group = [self mediaGroupWithThreadId:thread.uniqueId];
+    return [[self galleryExtensionWithTransaction:transaction] numberOfItemsInGroup:group];
+}
+
+- (NSUInteger)mediaIndexForMessage:(TSMessage *)message transaction:(YapDatabaseReadTransaction *)transaction
+{
+    NSString *groupId;
+    NSUInteger index;
```

Comment:
What will these values be if the message if no longer exists, e.g. it "disappeared"?

Reply:
Keep in mind our transaction is from a long-lived connection, so this shouldn't happen, but it's good point.

I've added an assert to verify this doesn't happen.